synapse/handlers/e2e_keys.py

```diff
@@ -977,12 +985,66 @@ def _get_e2e_cross_signing_verify_key(self, user_id, key_type, from_user_id=None
         Raises:
             NotFoundError: if the key is not found
         """
+        user = UserID.from_string(user_id)
```

Comment:
at this point I suspect `user_id` may be unvalidated input, so this can raise a SynapseError if the user_id is invalid. 

Although I consider it an antipattern for deeply nested functions like this to determine what the HTTP error code should be, I can't think of any immediately better solutions. Still, could you add it to the `Raises` list and that of any callers that don't catch it?

Reply:
Looks like all callers catch `SynapseError`s.